edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentProtocolHead.cs

```diff
@@ -95,13 +95,15 @@ public async Task CloseAsync(CancellationToken token)
         }
 
         public void Dispose()
+        {
+            DisposeAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
How do we decide whether or not to wrap wrap in Task.Run? From what I looked up wrapping in Task.Run() is likely to block a threadpool thread but will not result in async deadlock. I don't really understand this though.

Reply:
https://stackoverflow.com/questions/59018541/when-to-use-task-run-getawaiter-getresult-and-getawaiter-getresult